test/int/deployment_test.go

```diff
@@ -460,6 +460,36 @@ var _ = Describe("AtlasDeployment", Label("int", "AtlasDeployment"), func() {
 			})
 		})
 
+		It("Should Success (AWS) with enabled autoscaling", func() {
+			createdDeployment = mdbv1.DefaultAWSDeployment(namespace.Name, createdProject.Name)
+			createdDeployment.Spec.DeploymentSpec.AutoScaling = &mdbv1.AutoScalingSpec{
+				AutoIndexingEnabled: boolptr(true),
+				DiskGBEnabled:       boolptr(true),
+			}
+
+			By(fmt.Sprintf("Creating the Deployment %s with autoscaling", kube.ObjectKeyFromObject(createdDeployment)), func() {
+				Expect(k8sClient.Create(context.Background(), createdDeployment)).ToNot(HaveOccurred())
+
+				Eventually(testutil.WaitFor(k8sClient, createdDeployment, status.TrueCondition(status.ReadyType), validateDeploymentCreatingFunc()),
+					DeploymentUpdateTimeout, interval).Should(BeTrue())
+
+				doRegularDeploymentStatusChecks()
+				checkAtlasState()
+			})
+
+			By("Decreasing the Deployment disk size", func() {
+				createdDeployment.Spec.DeploymentSpec.DiskSizeGB = intptr(12)
+				performUpdate(30 * time.Minute)
+				doRegularDeploymentStatusChecks()
+				checkAtlasState(func(c *mongodbatlas.Cluster) {
+					Expect(*c.DiskSizeGB).To(BeEquivalentTo(*createdDeployment.Spec.DeploymentSpec.DiskSizeGB))
+
+					// check whether https://github.com/mongodb/go-client-mongodb-atlas/issues/140 is fixed
```

Comment:
Is this comment still relevant?

Reply:
Yes, but we have a different issue for this one. It will be removed later.